pshtt.py

```diff
@@ -402,29 +418,30 @@ def canonical_endpoint(http, httpwww, https, httpswww):
     # It allows a site to be canonically HTTPS if the cert has
     # a valid hostname but invalid chain issues.
 
-    is_https = (
-        (
-            (https.live and (not https.https_bad_hostname)) or
-            (httpswww.live and (not https.https_bad_hostname))
```

Comment:
I refactored this whole section on detecting whether the domain is canonically HTTPS, but you can see in the line above that there was a bug that prompted this, where I was checking `https.https_bad_hostname` for both `https` and `httpswww`, which was a bug and was causing issues. Refactoring this enabled me to more easily track down and fix this inconsistency.